### PR TITLE
Clarify code checking for UMD exports... 

### PR DESCRIPTION
**Pull request by andy-ms**
…and eagerly return `undefined` rather than continuing on to the for loop.

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -17713,14 +17713,17 @@ namespace ts {
                 const parentSymbol = getParentOfSymbol(symbol);
                 if (parentSymbol) {
                     if (parentSymbol.flags & SymbolFlags.ValueModule && parentSymbol.valueDeclaration.kind === SyntaxKind.SourceFile) {
+                        const symbolFile = <SourceFile> parentSymbol.valueDeclaration;
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts]: No space after type assertions (@weswigham can you write this rule?)

Comment (weswigham) [inline src/compiler/checker.ts]: I could. ;)

File: src/compiler/checker.ts [hunk 2]
```diff
@@ -17713,14 +17713,17 @@ namespace ts {
...
-                        if (parentSymbol.valueDeclaration === getSourceFileOfNode(node)) {
-                            return <SourceFile>parentSymbol.valueDeclaration;
-                        }
+                        const symbolIsUmdExport = symbolFile !== referenceFile;
+                        return symbolIsUmdExport ? undefined : symbolFile;
                     }
-                    for (let n = node.parent; n; n = n.parent) {
-                        if ((n.kind === SyntaxKind.ModuleDeclaration || n.kind === SyntaxKind.EnumDeclaration) && getSymbolOfNode(n) === parentSymbol) {
-                            return <ModuleDeclaration | EnumDeclaration>n;
+                    else {
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts]: What does this `else` add apart from nesting?

File: src/compiler/checker.ts [hunk 3]
```diff
@@ -17713,14 +17713,17 @@ namespace ts {
...
+                        for (let n = node.parent; n; n = n.parent) {
+                            if ((n.kind === SyntaxKind.ModuleDeclaration || n.kind === SyntaxKind.EnumDeclaration) && getSymbolOfNode(n) === parentSymbol) {
+                                return <ModuleDeclaration | EnumDeclaration>n;
+                            }
                         }
                     }
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts]: Explicitly return `undefined`

File: src/compiler/checker.ts (hunk 2 above)
Comment (andy-ms) [inline src/compiler/checker.ts]: The old version of this code fell through when it shouldn't have, so as part of removing the fallthrough I turned it into an if-else. If you think the nesting is too much I'll change it back.

**Comment by mhegazy** [comment]
👍

**Pull request merged**